### DISCUSSION TITLE
Mesh.ClosestPoint()

### DISCUSSION THREAD

**Discussion by iltabe** [cesarecaoduro/GeometrySharp#415] [General]
Hi guys!

I know some of you for different reasons! Ciao!

For a client I'm building an algorithm that requires some geometrical operations. I've looked into G-Shark and started to use it. It looks promising and does almost everything I need (basic vector, point, plane stuff).
Thanks guys for having give birth to this nice project!

The only thing I miss is a `mesh.ClosestPoint(Point3 pt)` method... Is it in the todo list by any luck?
Thanks!
Gianluca

**Comment by d3ssy** [comment]
Ciao Gianluca! Thanks for the kind words... 

Mesh is only a data structure / representation at the moment... no operations yet, but closest point is a fairly easy one to implement.

What is your expected behavior for this function? Return nearest mesh vertex to the input point? Or return the nearest point **on a mesh face** closest to the input point? 

Feel free to open an issue for this outlining the expected functionality.

**Comment by iltabe** [comment]
Thank you Guido!

Unfortunately the closest point implementation I need is a full implementation (see issue #416).
The mesh in my project will represent the ground of a large building site. The definition of it could vary from a very detailed mesh with lots of points to just a planar mesh with few very big faces. Refining the mesh in very small faces and - therefor - lot of points it is not really an option for me.

Happy to contribute for what I can. I can be the tester for sure.
